kiwi/volume_manager/btrfs.py

```diff
@@ -347,9 +355,9 @@ def _create_snapshot_info(self, filename):
 
     def _get_subvol_name_from_mountpoint(self, volume_mount):
         subvol_name = '/'.join(volume_mount.mountpoint.split('/')[3:])
-        if self.custom_args['root_is_snapshot']:
-            subvol_name = subvol_name.replace('.snapshots/1/snapshot/', '')
-        return os.path.normpath(subvol_name)
+        if self.toplevel_volume and self.toplevel_volume in subvol_name:
```

Comment:
Yes I was in doubt about that also. The '@' is included because the toplevel_volume always includes it (in fact it can only be '@' or '@/.snapshots/1/snapshot'), so it gets replaced if the mountpoint already includes the toplevel subvolume path. Then '@' is included out of the if clause because in case the mountpoint does not include the toplevel subvolume it means it in the shape of something line `/tmp/tmpdirxxx/var/cache`, so it that case it also needs to be appended.

I know this lines are quite ofuscated, I don't like them either, I tried to keep them as simple as possible, not to make a complex  `_get_subvol_name_from_mountpoint` method.